src/together/utils/files.py

```diff
@@ -96,6 +96,168 @@ def check_file(
+                )
+            else:
+                if not isinstance(message[column], str):
+                    raise InvalidFileFormatError(
+                        message=f"Invalid format on line {idx + 1} in the column {column} for turn `{message}` "
+                        f"of the input file. Expected string. Found {type(message[column])}",
+                        line_number=idx + 1,
+                        error_source="text_field",
+                    )
+
+        if has_weights and "weight" in message:
+            weight = message["weight"]
+            if not isinstance(weight, int):
+                raise InvalidFileFormatError(
+                    message="Weight must be an integer",
+                    line_number=idx + 1,
+                    error_source="key_value",
+                )
+            if weight not in {0, 1}:
+                raise InvalidFileFormatError(
+                    message="Weight must be either 0 or 1",
+                    line_number=idx + 1,
+                    error_source="key_value",
+                )
+        if message["role"] not in POSSIBLE_ROLES_CONVERSATION:
+            raise InvalidFileFormatError(
+                message=f"Found invalid role `{message['role']}` in the messages on the line {idx + 1}. "
+                f"Possible roles in the conversation are: {POSSIBLE_ROLES_CONVERSATION}",
+                line_number=idx + 1,
+                error_source="key_value",
+            )
+
+        if previous_role == message["role"]:
+            raise InvalidFileFormatError(
+                message=f"Invalid role turns on line {idx + 1} of the input file. "
+                "`user` and `assistant` roles must alternate user/assistant/user/assistant/...",
+                line_number=idx + 1,
+                error_source="key_value",
+            )
+        previous_role = message["role"]
+
+    return messages, has_weights
+
+
+def validate_preference_openai(example: Dict[str, Any], idx: int = 0) -> Dict[str, Any]:
+    """Validate the OpenAI preference dataset format.
+
+    Args:
+        example (dict): Input entry to be checked.
+        idx (int): Line number in the file.
+
+    Raises:
+        InvalidFileFormatError: If the dataset format is invalid.
+
+    Returns:
+        Dict[str, Any]: The validated example.
+    """
+    if not isinstance(example["input"], dict):
+        raise InvalidFileFormatError(
+            message="The dataset is malformed, the `input` field must be a dictionary.",
+            line_number=idx + 1,
+            error_source="key_value",
+        )
+
+    if "messages" not in example["input"]:
+        raise InvalidFileFormatError(
+            message="The dataset is malformed, the `input` dictionary must contain a `messages` field.",
+            line_number=idx + 1,
+            error_source="key_value",
+        )
+
+    example["input"]["messages"], _ = validate_messages(
+        example["input"]["messages"], idx
+    )
```

Comment:
We don't modify anything in messages, I would simply make `validate_messages` return nothing and raise an exception in case of an error